tests/data_access/models/test_user_model.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+
+import src.data_access.models.user_model as user_model
+from src.data_access.models.user_model import User
+
+
+class TestUserPassword:
+    def test_throws_error_when_trying_to_access_password(self):
+        user = User("username", "password")
+        with pytest.raises(AttributeError):
+            user.password
+
+    def test_calls_generate_password_hash_on_given_password(self, mocker):
+        expected_hash = "hash"
+        mocker.patch.object(
+            user_model, "generate_password_hash", return_value=expected_hash
+        )
+        user = User("username", "password")
+        actual = user.password_hash
+        assert actual == expected_hash
```

Comment:
here too, the return value doesnt affect.. u can save the mock in a variable, for example mock_generate_password_hash and use mock_generate_password_hash.return_value